### DISCUSSION TITLE
How to use hidden import when the package name is different from import.

### DISCUSSION THREAD

**Discussion by music-cat-bread** [pyinstaller/pyinstaller#6721] [Help]
Hello.

I installed a package called `pypiwin32` using pip, but import statement is `from win32crypt import CryptUnprotectData`, and `--hidden-import` don't work with `win32crypt.CryptUnprotectData` / `pypiwin32.win32crypt.CryptUnprotectData`.
So, how do I do that?

**Comment by rokm** [comment]
Why do you need a hidden import? Freezing a basic program:

```python
from win32crypt import CryptUnprotectData
print(CryptUnprotectData)
```

collects `site-packages\win32\win32crypt.pyd` into `win32crypt.pyd`, so the import should work in the frozen program.

Originally, it is located in the `win32` sub-directory, but `site-packages\pywin32.pth` adds `win32` sub-directory (among others) to the search path. 

So if anything, the hidden import should be just `win32crypt` (`CryptUnprotectData` is a symbol within that extension module).

**Comment by music-cat-bread** [comment]
But then I run my program it says:
`ModuleNotFoundError: No module named 'win32crypt'`
I think it have to do something with different name than that on pip because I don't have similar problems with any other package.
Also I forget to mention this import is not visible in script and is hidden.

**Reply by rokm** [reply]
Ah, okay, if that import occurs in a compiled extension, or if it is done indirectly (lazy import, or programmatic import using functional form of import), then yes, you will need a hidden import.

Try using `win32crypt`, then. If that does not work, provide a minimum example that reproduces the problem and I'll check what's going on.

**Reply by rokm** [reply]
Or if you copy & paste the whole traceback, so we can see where the error originates from...

**Reply by music-cat-bread** [reply]
Okey, so as I was getting all the files and getting this comment ready to sent the error disappiered.
I really don't know what was the issue. I ran couple more tests but all were successfull.

Sorry, for wasting a little bit of your time.
